e2e-polybft/framework/test-cluster.go

```diff
@@ -462,23 +470,24 @@ func NewTestCluster(t *testing.T, validatorsCount int, opts ...ClusterOption) *T
 	}
 
 	for i := 1; i <= int(cluster.Config.ValidatorSetSize); i++ {
-		cluster.InitTestServer(t, i, true, cluster.Config.HasBridge && i == 1 /* relayer */)
+		cluster.InitTestServer(t, i, cluster.Config.ValidatorPrefix,
+			true, cluster.Config.HasBridge && i == 1 /* relayer */)
 	}
 
 	for i := 1; i <= cluster.Config.NonValidatorCount; i++ {
-		offsetIndex := i + int(cluster.Config.ValidatorSetSize)
-		cluster.InitTestServer(t, offsetIndex, false, false /* relayer */)
+		cluster.InitTestServer(t, i, nonValidatorPrefix, false, false /* relayer */)
 	}
 
 	return cluster
 }
 
-func (c *TestCluster) InitTestServer(t *testing.T, i int, isValidator bool, relayer bool) {
+func (c *TestCluster) InitTestServer(t *testing.T, i int,
+	validatorPrefix string, isValidator bool, relayer bool) {
```

Comment:
Why not providing data directory constructed from the outside?

```suggestion
func (c *TestCluster) InitTestServer(t *testing.T,
	dataDir string, isValidator bool, relayer bool) {
```

Reply:
Fix: https://github.com/0xPolygon/polygon-edge/pull/1378/commits/e796176e70d7865daae6f3b1e410e8856678e398